index.js

```diff
@@ -39,184 +39,158 @@ module.exports.s3 = function(config, key, args, callback) {
-							});
-						}
-
-						// apply the resize
-						args.resize = getDimArray( args.resize, zoom );
-						image.resize({
-							width: args.resize[0],
-							height: args.resize[1],
-							withoutEnlargement: true,
-							position: ( args.crop_strategy !== 'smart' && args.crop_strategy ) || args.gravity || 'centre',
-						});
-					} else if (args.fit) {
-						args.fit = getDimArray( args.fit, zoom );
-						image.resize({
-							width: args.fit[0],
-							height: args.fit[1],
-							fit: 'inside',
-							withoutEnlargement: true,
-						});
-					} else if (args.lb) {
-						args.lb = getDimArray( args.lb, zoom );
-						image.resize({
-							width: args.lb[0],
-							height: args.lb[1],
-							fit: 'contain',
-							// default to a black background to replicate Photon API behaviour
-							// when no background colour specified
-							background: args.background || 'black',
-							withoutEnlargement: true,
-						});
-					} else if (args.w || args.h) {
-						image.resize({
-							width: (Number(args.w) * zoom) || null,
-							height: (Number( args.h ) * zoom) || null,
-							fit: args.crop ? 'cover' : 'inside',
-							withoutEnlargement: true,
-						});
-					}
-
-					// return a default compression value based on a logarithmic scale
-					// defaultValue = 100, zoom = 2; = 65
-					// defaultValue = 80, zoom = 2; = 50
-					// defaultValue = 100, zoom = 1.5; = 86
-					// defaultValue = 80, zoom = 1.5; = 68
-					var applyZoomCompression = function( defaultValue, zoom ) {
-						return clamp( Math.round( defaultValue - ( (Math.log(zoom) / Math.log(defaultValue / zoom)) * (defaultValue * zoom) ) ), Math.round(defaultValue / zoom), defaultValue );
-					}
-
-					// set default quality slightly higher than sharp's default
-					if ( ! args.quality ) {
-						args.quality = applyZoomCompression( 82, zoom );
-					}
-
-					// allow override of compression quality
-					if (args.webp) {
-						image.webp({
-							quality: Math.round( clamp( args.quality, 0, 100 ) ),
-						});
-					} else if (metadata.format === 'jpeg') {
-						image.jpeg({
-							quality: Math.round( clamp( args.quality, 0, 100 ) ),
-						});
-					}
-
-					// send image
-					return image.toBuffer(function(err, _data, info) {
-						if (err) {
-							reject(err);
-							if (callback) {
-								callback(err);
-							}
-							return;
-						}
-
-						resolve({ data: _data, info: info });
-						if (callback) {
-							callback(err, _data, info);
-						}
-						return;
-					});
-				});
+// return a default compression value based on a logarithmic scale
+// defaultValue = 100, zoom = 2; = 65
+// defaultValue = 80, zoom = 2; = 50
+// defaultValue = 100, zoom = 1.5; = 86
+// defaultValue = 80, zoom = 1.5; = 68
+const applyZoomCompression = function( defaultValue, zoom ) {
+	return clamp( Math.round( defaultValue - ( (Math.log(zoom) / Math.log(defaultValue / zoom)) * (defaultValue * zoom) ) ), Math.round(defaultValue / zoom), defaultValue );
```

Comment:
Spacin' needs to be consistent here

Also, maybe wrap the lines?

Reply:
Cheers, fixed this one.